nixos/modules/tasks/network-interfaces.nix

```diff
@@ -190,9 +190,11 @@ let
         type = types.nullOr types.bool;
         default = null;
         description = lib.mdDoc ''
-          Whether this interface should be configured with dhcp.
-          Null implies the old behavior which depends on whether ip addresses
-          are specified or not.
+          Whether this interface should be configured with DHCP. Overrides the
+          default set by {option}`networking.useDHCP`. If `null` (the default),
+          DHCP is enabled if the interface has any IPv4 addresses configured
```

Comment:
I think that's the exact opposite of what's intended?

I think "DHCP is enabled if the interface __doesn't__ have any IPv4 addresses configured" is what you mean.

Reply:
Oops, fixed!